src/components/CheckboxButton.js

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
```

Comment:
Now Why do we need this file?

Reply:
We need this file because checkbox behavior (without check icon) is also used in `BaseTextInput` for secure text (eye icon).

Reply:
I just tested on the native device and it is not working there. Sorry about that I broke it while refactoring. `event.type` is not `click` on native so it always returns early and never calls onPress. 

We also need to add native implementation `CheckboxButton/index.native.js`

Reply:
I am a little confused here.

> Ok, then let make changes to CheckBox.js file to be usable everywhere and merge this CheckboxButton into Checkbox.

Do you mean to import CheckboxButton in Checkbox and use it there? If so it is already done that way and if we need some where else Checkbox behavior we can import and use CheckboxButton eg: BaseTextInput

Reply:
I meant to say merge checkBox.js and CheckboxButton into a single component. I see that Checkbox is just an extra wrapper and does not do much. Let me know otherwise.

Reply:
We have to pass the icon as the prop then. Default will be check mark as it is. We will pass the icon from BaseTextInput for secure input (eye icon).

Reply:
I think you can just use `props.children`. When pass used that instead of checkBox children of Pressable.

Reply:
I have pushed the changes and it is not working on native device as explained [here](https://github.com/Expensify/App/pull/8874#discussion_r893506108). Implementing on native will be a lot of duplicate code. How do you suggest moving ahead?

Reply:
It can be done by modifying the condition or I am missing something.

Reply:
what if `event && event.type !== 'click'`.

Reply:
Yes, we can do this way. Thanks, updated the code.